### PR TITLE
feat: #153291262 - save message storage preference

### PR DESCRIPTION
**Pull request by gunzip**

### CODE REVIEW

File: api/definitions.yaml [hunk 1]
```diff
@@ -153,6 +153,11 @@ ExtendedProfile:
       $ref: "#/EmailAddress"
     preferred_languages:
       $ref: "#/PreferredLanguages"
+    is_storage_of_message_content_enabled:
```
Comment (cloudify) [inline api/definitions.yaml]: I would use a shorter name, like `inbox_enabled`, "whether the user enabled the inbox feature, for details on how the inbox feature works see..."

File: lib/api/admin_api.ts [hunk 2]
```diff
@@ -417,6 +417,12 @@ export const specs = {
       properties: {
         email: { $ref: "#/definitions/EmailAddress" },
         preferred_languages: { $ref: "#/definitions/PreferredLanguages" },
+        is_storage_of_message_content_enabled: {
```
Comment (cloudify) [inline lib/api/admin_api.ts]: `$ref` here too?

File: lib/api/public_api_v1.ts [hunk 3]
```diff
@@ -380,6 +380,12 @@ export const specs = {
       properties: {
         email: { $ref: "#/definitions/EmailAddress" },
         preferred_languages: { $ref: "#/definitions/PreferredLanguages" },
+        is_storage_of_message_content_enabled: {
```
Comment (cloudify) [inline lib/api/public_api_v1.ts]: `$ref` to definitions?

File: lib/controllers/profiles.ts [hunk 4]
```diff
@@ -59,7 +59,9 @@ import { ServiceModel } from "../models/service";
 function toExtendedProfile(profile: RetrievedProfile): ExtendedProfile {
   return {
     email: profile.email,
-    version: profile.version
+    version: profile.version,
+    is_storage_of_message_content_enabled:
+      profile.isStorageOfMessageContentEnabled
```
Comment (cloudify) [inline lib/controllers/profiles.ts]: again I would give it a feature name, like `inbox` -> `isInboxEnabled`

"Inbox" may not be only about storing messages, the feature may provide other functionalities (e.g. search?), so it makes sense to use a name that is decoupled from a specific functionality

**Comment by cloudify** [comment]
I'd like to merge PR #21 before this one

File: lib/api/admin_api.ts (hunk 2 above)
Comment (gunzip) [reply lib/api/admin_api.ts]: About using $ref or embedding the definition, actually there's no clear rule. Sometimes we use $ref(s) some others we embed the definitions (ie. CreatedMessage -> sender_service_id).

Comment (cloudify) [reply lib/api/admin_api.ts]: when in doubt I would prefer avoiding duplication of definitions as they may go out of sync

**Comment by cloudify** [comment]
linting errors, likely due to PR #121

**Pull request merged**